Sources/TensorFlow/Loss.swift

```diff
@@ -86,6 +86,27 @@ public func categoricalHingeLoss<Scalar: TensorFlowFloatingPoint>(
     return max(Tensor(0), negative - positive + Tensor(1))
 }
 
+/// Helper function for Logcosh
+@differentiable(wrt: x)
+internal func logcosh<Scalar: TensorFlowFloatingPoint>(
+    x: Tensor<Scalar>
+) -> Tensor<Scalar> {
+    let y = Tensor<Scalar>([2])
+    return  x + softplus(Tensor(-2) * x) - log(y)
+}
+
+/// Returns the Logcosh loss between predictions and expectations.
+///
+/// - Parameters:
+///   - predicted: Predicted outputs from a neural network.
+///   - expected: Expected values, i.e. targets, that correspond to the correct output.
+@differentiable(wrt: predicted)
+public func logcoshLoss<Scalar: TensorFlowFloatingPoint>(
```

Comment:
Please rename using camelcase:
```suggestion
public func logCoshLoss<Scalar: TensorFlowFloatingPoint>(
```